src/solvers/dgmulti/dg.jl

```diff
@@ -8,16 +8,29 @@
 # out <- A*x
 mul_by!(A) = @inline (out, x)->matmul!(out, A, x)
 
-# specialize for SBP operators since `matmul!` doesn't work for `UniformScaling` types.
-mul_by!(A::UniformScaling) = @inline (out, x)->mul!(out, A, x)
-mul_by_accum!(A::UniformScaling) = @inline (out, x)->mul!(out, A, x, one(eltype(out)), one(eltype(out)))
-
 # out <- out + A * x
 mul_by_accum!(A) = @inline (out, x)->matmul!(out, A, x, one(eltype(out)), one(eltype(out)))
 
 #  out <- out + α * A * x
 mul_by_accum!(A, α) = @inline (out, x)->matmul!(out, A, x, α, one(eltype(out)))
 
+# specialize for SBP operators since `matmul!` doesn't work for `UniformScaling` types.
+struct MulByUniformScaling end
+struct MulByAccumUniformScaling end
+mul_by!(A::UniformScaling) = MulByUniformScaling()
+mul_by_accum!(A::UniformScaling) = MulByAccumUniformScaling()
+
+# StructArray fallback
+@inline apply_to_each_field(f::Function, args...) = StructArrays.foreachfield(f, args...)
```

Comment:
```suggestion
@inline apply_to_each_field(f, args...) = StructArrays.foreachfield(f, args...)
```
Otherwise, it's overly restrictive, which shouldn't be necessary here.